### ISSUE TITLE
matlab table causes trouble: zlib error: incorrect header check (code: -3)

### ISSUE THREAD

**Issue by StefanPofahl** [JuliaIO/HDF5.jl#915]
First of all thank you for this great tool and for the effort to make matlab-tables accessible!

I have a mat-file, containing a matlab-table.
If use the command `names()` (which is deprecated) it works,
but if I use `keys()` instead it throws an error:
`hd_mat_file = matopen("filename.mat")
variable_keys = keys(hd_mat_file) 
variable_names = names(hd_mat_file)
`
Here part of the error-message:
`zlib error: incorrect header check (code: -3)
changemode!(stream::TranscodingStreams.TranscodingStream{CodecZlib.ZlibDecompressor, IOStream}, newmode::Symbol) at stream.jl:717
callprocess(stream::TranscodingStreams.TranscodingStream{CodecZlib.ZlibDecompressor, IOStream}, inbuf::TranscodingStreams.Buffer, outbuf::TranscodingStreams.Buffer) at stream.jl:649
fillbuffer(stream::TranscodingStreams.TranscodingStream{CodecZlib.ZlibDecompressor, IOStream}; eager::Bool) at stream.jl:577
fillbuffer at stream.jl:564 [inlined]
[...]
`
A curiosity is on my machine (MS Windows 10, Julia v1.7.2):
If I call first `names()` and afterwards `keys()` it works (in the line-by-line-execution mode).

Regards,
Stefan

[Table_PolCurve.zip](https://github.com/JuliaIO/HDF5.jl/files/8261446/Table_PolCurve.zip)

**Comment by musm** [comment]
It looks like you are using `MAT.jl` ? 
Which package versions are you using?

**Comment by StefanPofahl** [comment]
Yes, MAT v0.10.3, HDF5 v0.16.3, the behavior is the same on Linux and MS Windows 10.

**Comment by tn9702135** [comment]
hi 
I have an error to read() the similar it.  I did it but it did work.
my error
LoadError: zlib error: incorrect header check (code: -3)
in expression starting at C:\Users\ECE_CPU_Node_20\Desktop\Dr_Forrester_network\PhD-Codes-master\Chapter-6\Delay_timeseries.jl:22
changemode!(stream::TranscodingStreams.TranscodingStream{CodecZlib.ZlibDecompressor, IOStream}, newmode::Symbol) at stream.jl:717
callprocess(stream::TranscodingStreams.TranscodingStream{CodecZlib.ZlibDecompressor, IOStream}, inbuf::TranscodingStreams.Buffer, outbuf::TranscodingStreams.Buffer) at stream.jl:649
fillbuffer(stream::TranscodingStreams.TranscodingStream{CodecZlib.ZlibDecompressor, IOStream}; eager::Bool) at stream.jl:577
fillbuffer at stream.jl:564 [inlined]
eof(stream::TranscodingStreams.TranscodingStream{CodecZlib.ZlibDecompressor, IOStream}) at stream.jl:188
unsafe_read(stream::TranscodingStreams.TranscodingStream{CodecZlib.ZlibDecompressor, IOStream}, output::Ptr{UInt8}, nbytes::UInt64) at stream.jl:356
unsafe_read at io.jl:724 [inlined]
unsafe_read(s::TranscodingStreams.TranscodingStream{CodecZlib.ZlibDecompressor, IOStream}, p::Base.RefValue{UInt32}, n::Int64) at io.jl:723
read! at io.jl:725 [inlined]
read at io.jl:729 [inlined]
read_bswap at MAT_v5.jl:83 [inlined]
read_header at MAT_v5.jl:110 [inlined]
getvarnames(matfile::MAT.MAT_v5.Matlabv5File) at MAT_v5.jl:396
keys at MAT_v5.jl:415 [inlined]
names(matfile::MAT.MAT_v5.Matlabv5File) at MAT.jl:176
top-level scope at Delay_timeseries.jl:22
eval at boot.jl:373 [inlined]
include_string(mapexpr::typeof(identity), mod::Module, code::String, filename::String) at loading.jl:1196

**Comment by tn9702135** [comment]
hi 
I have an error to run line read() that error is below, I did your suggestion and using MAT v0.10.3, HDF5 v0.16.3.
but I did not work.

error: 
LoadError: zlib error: incorrect header check (code: -3)
in expression starting at C:\Users\ECE_CPU_Node_20\Desktop\Dr_Forrester_network\PhD-Codes-master\Chapter-6\Delay_timeseries.jl:22
changemode!(stream::TranscodingStreams.TranscodingStream{CodecZlib.ZlibDecompressor, IOStream}, newmode::Symbol) at stream.jl:717
callprocess(stream::TranscodingStreams.TranscodingStream{CodecZlib.ZlibDecompressor, IOStream}, inbuf::TranscodingStreams.Buffer, outbuf::TranscodingStreams.Buffer) at stream.jl:649
fillbuffer(stream::TranscodingStreams.TranscodingStream{CodecZlib.ZlibDecompressor, IOStream}; eager::Bool) at stream.jl:577
fillbuffer at stream.jl:564 [inlined]
eof(stream::TranscodingStreams.TranscodingStream{CodecZlib.ZlibDecompressor, IOStream}) at stream.jl:188
unsafe_read(stream::TranscodingStreams.TranscodingStream{CodecZlib.ZlibDecompressor, IOStream}, output::Ptr{UInt8}, nbytes::UInt64) at stream.jl:356
unsafe_read at io.jl:724 [inlined]
unsafe_read(s::TranscodingStreams.TranscodingStream{CodecZlib.ZlibDecompressor, IOStream}, p::Base.RefValue{UInt32}, n::Int64) at io.jl:723
read! at io.jl:725 [inlined]
read at io.jl:729 [inlined]
read_bswap at MAT_v5.jl:83 [inlined]
read_header at MAT_v5.jl:110 [inlined]
getvarnames(matfile::MAT.MAT_v5.Matlabv5File) at MAT_v5.jl:396
keys at MAT_v5.jl:415 [inlined]
names(matfile::MAT.MAT_v5.Matlabv5File) at MAT.jl:176
top-level scope at Delay_timeseries.jl:22
eval at boot.jl:373 [inlined]
include_string(mapexpr::typeof(identity), mod::Module, code::String, filename::String) at loading.jl:1196

**Comment by mkitti** [comment]
This is really a MAT.jl issue. In the meantime, does `matread` work per https://github.com/JuliaIO/MAT.jl/issues/80#issuecomment-1219575773
